config/webpack.config.js

```diff
@@ -11,10 +11,59 @@ var LodashModuleReplacementPlugin = require('lodash-webpack-plugin');
 var pluginUtils = require('../script/plugin_webpack_directories');
 var vendorEntry = require('./webpack.vendor');
 var SimpleNamedModulesPlugin = require('../webpack/simple_named_modules');
+var argvParse = require('argv-parse');
+var fs = require('fs');
+var os = require('os');
+var { execSync } = require('child_process');
+
+var args = argvParse({
+  https: {
+    type: 'boolean',
+  },
+  public: {
+    type: 'string',
+  },
+  port: {
+    type: 'string',
+  },
+  host: {
+    type: 'string',
+  }
+})
+
+const supportedLocales = () => {
+  const localeDir = path.join(__dirname, '..', 'locale');
+
+  // Find all files in ./locale/*
+  const localesFiles = fs.readdirSync(localeDir)
+
+  // Return only folders
+  return localesFiles.filter(f => fs.statSync(path.join(localeDir, f)).isDirectory());
+}
+
+const supportedLanguages = () => {
+  // Extract extract languages from the language tags (strip off dialects)
+  return [ ...new Set(supportedLocales().map(d => d.split('_')[0]))];
+}
+
+const removePort = host => host && host.replace(/:[0-9]*$/, "");
+
+const devServerConfig = () => {
+  const result = require('dotenv').config();
+  if (result.error && result.error.code !== 'ENOENT') {
+    throw result.error;
+  }
+
+  return {
```

Comment:
can we extract this to a different file ? (not to clutter this main config file).

Reply:
I'd prefer to keep it here to have all the configuration at one place. If it starts growing in future, we can revisit it.